### DISCUSSION TITLE
Using Layers with imports and backlinks

### DISCUSSION THREAD

**Discussion by Freeskier73** [terrastruct/d2#2136] [Q&A] [answered]
How do I split layers to different files while still using backlinks.
Example that works:
```
a -> b
b -> c
b -> d

c.link: layers.c
d.link: layers.d

layers: {
  c: {
    e -> f
    f -> g
    e.link: _
  }

  d: { 
    e -> h
    h -> i
    e.link: _
  }
}
```
When I try to move one layer to a separate file, I get an error:
err: failed to compile Sample.d2: SampleC.d2:3:1: invalid underscore usage

Sample.d2:
```
a -> b
b -> c
b -> d

c.link: layers.c
d.link: layers.d

layers: {
  c: @SampleC

  d: { 
    e -> h
    h -> i
    e.link: _
  }
}
```

SampleC.d2:
```
e -> f
f -> g
e.link: _
```
Is this not possible or do I have to use another syntax.
I also tried defining `e.link` in the layer definition in Sample.d2, but this results in an additional box containing the link...

**Comment by alixander** [comment, marked as answer]
@Freeskier73 What version are you on? This should work in the latest version

**Reply by Freeskier73** [reply]
I´ve been on 0.6.6...
With 0.6.7 it works as I thought it should.

Thank you very much!

**Reply by alixander** [reply]
🍻
